util/oteltest/src/oteltest/private.py

```diff
@@ -0,0 +1,283 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import glob
+import importlib
+import inspect
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+import time
+import typing
+import venv
+from pathlib import Path
+
+from google.protobuf.json_format import MessageToDict
+from oteltest import OtelTest, telemetry
+from oteltest.sink import GrpcSink, RequestHandler
+
+from opentelemetry.proto.collector.logs.v1.logs_service_pb2 import (
+    ExportLogsServiceRequest,
+)
+from opentelemetry.proto.collector.metrics.v1.metrics_service_pb2 import (
+    ExportMetricsServiceRequest,
+)
+from opentelemetry.proto.collector.trace.v1.trace_service_pb2 import (
+    ExportTraceServiceRequest,
+)
+
+
+def run(script_path: str, wheel_file: str, venv_parent_dir: str):
+    temp_dir = venv_parent_dir or tempfile.mkdtemp()
+    print(f"- Using temp dir for venvs: {temp_dir}")
+
+    if os.path.isdir(script_path):
+        handle_dir(script_path, temp_dir, wheel_file)
+    elif os.path.isfile(script_path):
+        handle_file(script_path, temp_dir, wheel_file)
+    else:
+        print(f"- {script_path} does not exist")
+        return
+
+
+def handle_dir(script_path, temp_dir, wheel_file):
+    sys.path.append(script_path)
+    for script in ls_scripts(script_path):
+        print(f"- Setting up environment for dir {script}")
+        setup_script_environment(temp_dir, script_path, script, wheel_file)
+
+
+def handle_file(script_path, temp_dir, wheel_file):
+    print(f"- Setting up environment for file {script_path}")
+    script_dir = os.path.dirname(script_path)
+    sys.path.append(script_dir)
+    setup_script_environment(
+        temp_dir, script_dir, os.path.basename(script_path), wheel_file
+    )
+
+
+def ls_scripts(script_dir):
+    original_dir = os.getcwd()
+    os.chdir(script_dir)
+    scripts = glob.glob("*.py")
+    os.chdir(original_dir)
+    return scripts
+
+
+def setup_script_environment(venv_parent, script_dir, script, wheel_file):
+    handler = AccumulatingHandler()
+    sink = GrpcSink(handler)
+    sink.start()
+
+    module_name = script[:-3]
+    oteltest_instance: OtelTest = load_test_class_for_script(module_name)()
+
+    script_venv = Venv(str(Path(venv_parent) / module_name))
+    script_venv.create()
+
+    pip_path = script_venv.path_to_executable("pip")
+
+    run_subprocess([pip_path, "install", (wheel_file or "oteltest")])
+
+    for req in oteltest_instance.requirements():
+        print(f"- Will install requirement: '{req}'")
+        run_subprocess([pip_path, "install", req])
+
+    stdout, stderr, returncode = run_python_script(
+        script_dir, script, oteltest_instance, script_venv
+    )
+    print_subprocess_result(stdout, stderr, returncode)
+
+    filename = get_next_json_file(script_dir, module_name)
```

Comment:
Would be nice to have an option to avoid saving these files

Reply:
Yes. Maybe should be off by default and on via a CLI switch. Noted for a later PR.